### PR TITLE
Remove the temporary gRPC log channel

### PR DESCRIPTION
**Pull request by Superskyyy**
- Removes the temporary log gRPC channel.
- Considers log filters, now log reporter won't report filtered out logs by default to avoid confusing the users.
  (New option to ignore log filters, defaults to `False`)
- Modifies all affected Docs.

I will implement `printing traceID/ context in logs` and `support for HTTP/ Kafka protocols` in future PRs.

Signed-off-by: YihaoChen <Superskyyy@outlook.com>

### CODE REVIEW

**Comment by Humbertzhang** [comment]
I will review this PR tonight.

**Review by Humbertzhang — CHANGES_REQUESTED** [review]

File: skywalking/agent/protocol/__init__.py [hunk 1]
```diff
@@ -35,5 +35,8 @@ def heartbeat(self):
     def report(self, queue: Queue, block: bool = True):
         raise NotImplementedError()
 
+    def report_log(self, queue: Queue, block: bool = True):
+        raise NotImplementedError()
```
[the remark this answers is not in the document]
Comment (kezhenxu94) [reply skywalking/agent/protocol/__init__.py:39]: We may want to provide some dummy implementations for http and kafka and log this kind of information only once, it's better we can implement those protocols later

[the remark this answers is not in the document]
Comment (tom-pytel) [reply skywalking/agent/protocol/__init__.py:39]: If it is documented that log reporting is not currently implemented for these protocols then it is the responsibility of the user to not enable log reporting if not using grpc, no?

**Review by tom-pytel — APPROVED** [review]

File: skywalking/agent/protocol/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Superskyyy) [reply skywalking/agent/protocol/__init__.py:39]: Well I plan to PR the HTTP and Kafka part tomorrow, shall we merge this one first?

[the remark this answers is not in the document]
Comment (tom-pytel) [reply skywalking/agent/protocol/__init__.py:39]: If @Humbertzhang agrees I have no problem merging as is if its going to be fixed tomorrow anyway.

[the remark this answers is not in the document]
Comment (Humbertzhang) [reply skywalking/agent/protocol/__init__.py:39]: @tom-pytel make sense! Let's merge this PR now.

[the remark this answers is not in the document]
Comment (Superskyyy) [reply skywalking/agent/protocol/__init__.py:39]: Thanks guys!

**Review by Humbertzhang — APPROVED** [review]

**Comment by wu-sheng** [comment]
@Superskyyy The main repo doc hasn't been updated yet, please fix it.

**Pull request merged**